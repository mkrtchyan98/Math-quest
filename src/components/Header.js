import React from 'react';
import { Link } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  root: {
        display:'none',
  [theme.breakpoints.down('sm')]: {
 background: 'transparent',
 display:'flex'
    },
    '& a':{
        textDecoration:'none',
        color:'#f97300',
        outline:'none'
    }
    }
}))



const Header = (props) => {
       const classes = useStyles();

    return (
            <header className={classes.root}>
            <div>
                <Link className="back-btn" to={"/"}>&larr;</Link>
                <Link to={"/"}>Arithmetic Questions</Link>
            </div>
        </header>
    )
}

export default Header