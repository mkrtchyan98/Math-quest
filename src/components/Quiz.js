import React, { useState } from 'react';
import Puzzle from './Puzzle';
import helpers from '../Helpers/Question';
import { makeStyles } from '@material-ui/core/styles';
import Header from './Header';
import Results from './Results';

const useStyles = makeStyles(theme =>({
  root: {
    display: 'flex',
    justifyContent:'center',
    alignItems:'center',
    flexDirection:'column',
    background:'black',
    width:'400px',
    height:'400px',
    backgroundImage:'linear-gradient(to right,rgba(152,90,21,.13), rgba(198, 153, 104, 0.17))',
    backgroundSize:'15px',
  [theme.breakpoints.down('sm')]: {
  maxWidth:'800px',
  width:'100%',
  margin:'0 auto',
  height:'100%',
  '& h1.Puzzle__Statement':{
    fontSize:'30px'
  }
  },
  '& span.Results__Symbol--Correct':{
    color:'green'
  },
  '& span.Results__Symbol--Incorrect':{
    color:'red'
  },
    '& div.Results': {
        display:'flex',
        flexDirection:'row',
        justifyContent:'end',
        alignItems:'center',
        position:'relative',
        right:'2px',

    },
    '& div.PuzzleContainer':{
        background:'#1e683d',
        width:'90%',
        height:'90%',
    },
    '& h2':{
        fontSize:'16px'
    },
    '& div.PuzzleResult':{
        display:'none'
    },
    '& div.QuestionNumber':{
        float:'left',
        fontSize:'calc(10px + 2vmin)',
        position:'relative',
        left:'2px',
    }
}
}));

const  Quiz = (props) =>  {
 const classes = useStyles();
const [counter,setCounter] = useState(0);
const [level,setLevel] = useState("single");
const [data,setData] = useState(helpers.Question("single",props.operator));
const [rightAnswers,setRightAnswers] = useState(0);
const [wrongAnswers,setWrongAnswers] = useState(0);
   const   choseLevel = (value) => {
        if (value > 5 && value <= 10) {
            setLevel("double")
        } else if (value > 10 && value <= 15) {
           setLevel("triple")
        } 
    }
   const  handleClick = (val) =>{
        if (val === data.correctAnswer) {
            setRightAnswers(rightAnswers+1)
        } else {
           setWrongAnswers(wrongAnswers+1)
        }
          setCounter(counter + 1);
          setData(helpers.Question(level,props.operator));
        if(counter > 5){
                  choseLevel(counter)
              }
        }    
        return (
            <div className={classes.root}>
            <Header  />
                <div className={counter === 15 ? "PuzzleResult" :"PuzzleContainer"}>
                <div className="QuestionNumber">
                {counter} of 15
                </div>
                    <div className="Results">
                        <div className="Results__Correct">
                            <span className="Results__Symbol--Correct">&#10004;</span>
                            <span className="Results__Count">{rightAnswers}</span>
                        </div>
                        <div className="Results__Incorrect">
                            <span className="Results__Symbol--Incorrect">&#x2718;</span>
                            <span className="Results__Count">{wrongAnswers}</span>
                        </div>
                    </div>
                    <Puzzle
                        handleClick={handleClick}
                        data={data} />
                </div>
                {
                    counter === 15 &&  
                    <Results rightAnswers={rightAnswers}  wrongAnswers={wrongAnswers} />
            }
            </div>
        );
 
}

export default Quiz