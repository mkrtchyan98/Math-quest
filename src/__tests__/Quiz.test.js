import Quiz from '../components/Quiz';
import React from 'react';
import { shallow, mount,render } from "enzyme";
import { MemoryRouter } from 'react-router-dom';
import exportFunctions from '../helpers/Question';
import Header from '../components/Header';

jest.mock('../helpers/Question')


	describe('<Quiz />',() => {
	test('renders snapshot',() => {
		const wrapper = shallow(<Quiz  operator="+"/>);
		expect(wrapper).toMatchSnapshot();
	})
	it('renders with props',()=>{
		const wrapper = mount (
					 <MemoryRouter initialEntries={[ {pathname:'/add'} ]} initialIndex={0}>
		<Quiz  operator="+"/>
		</MemoryRouter>)
		expect(wrapper.find(Header)).toHaveLength(1)
		exportFunctions.Question( "single","+")
		expect(exportFunctions.Question).toHaveBeenCalledWith("single","+")
		
	})

	})
