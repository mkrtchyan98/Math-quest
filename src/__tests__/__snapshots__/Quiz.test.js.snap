// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quiz /> renders snapshot 1`] = `ShallowWrapper {}`;
